src/game_interpreter_map.cpp

```diff
@@ -581,179 +581,171 @@ bool Game_Interpreter_Map::CommandPanScreen(RPG::EventCommand const& com) { // c
 	return true;
 }
 
-bool Game_Interpreter_Map::CommandShowPicture(RPG::EventCommand const& com) { // code 11110
-	int pic_id = com.parameters[0];
-	Game_Picture* picture = Main_Data::game_screen->GetPicture(pic_id);
-	std::string pic_name = com.string;
-	int x = ValueOrVariable(com.parameters[1], com.parameters[2]);
-	int y = ValueOrVariable(com.parameters[1], com.parameters[3]);
-	bool scrolls = com.parameters[4] > 0;
-	int magnify = com.parameters[5];
-	int top_trans = com.parameters[6];
-	bool use_trans = com.parameters[7] > 0;
-	int red = com.parameters[8];
-	int green = com.parameters[9];
-	int blue = com.parameters[10];
-	int saturation = com.parameters[11];
-	int effect = com.parameters[12];
-	int speed = com.parameters[13];
-	int bottom_trans;
-
-	// PicPointer Patch handling
+// PicPointerPatch handling.
+// See http://cherrytree.at/cms/download/?did=19
+namespace PicPointerPatch {
+
+static void AdjustId(int& pic_id) {
+	if (pic_id > 10000) {
+		int new_id;
+		if (pic_id > 50000) {
+			new_id = Game_Variables[pic_id - 50000];
+		} else {
+			new_id = Game_Variables[pic_id - 10000];
+		}
+
+		if (new_id > 0) {
+			Output::Debug("PicPointer: ID %d replaced with ID %d", pic_id, new_id);
+			pic_id = new_id;
+		}
+	}
+}
+
+static void AdjustParams(Game_Picture::Params& params) {
+	if (params.magnify > 10000) {
+		int new_magnify = Game_Variables[params.magnify - 10000];
+		Output::Debug("PicPointer: Zoom %d replaced with %d", params.magnify, new_magnify);
+		params.magnify = new_magnify;
+	}
+
+	if (params.top_trans > 10000) {
+		int new_top_trans = Game_Variables[params.top_trans - 10000];
+		Output::Debug("PicPointer: Top transparency %d replaced with %d", params.top_trans, new_top_trans);
+		params.top_trans = new_top_trans;
+	}
+
+	if (params.bottom_trans > 10000) {
+		int new_bottom_trans = Game_Variables[params.bottom_trans - 10000];
+		Output::Debug("PicPointer: Bottom transparency %d replaced with %d", params.bottom_trans, new_bottom_trans);
+		params.bottom_trans = new_bottom_trans;
+	}
+}
+
+static void AdjustShowParams(int& pic_id, Game_Picture::ShowParams& params) {
+	AdjustId(pic_id);
+	AdjustParams(params);
+
+	// Adjust name
 	if (pic_id >= 50000) {
```

Comment:
Bug is probably here because AdjustId overwrites the pic_id to the real number, so this check fails.